presto-accumulo/src/main/java/com/facebook/presto/accumulo/index/IndexLookup.java

```diff
@@ -0,0 +1,425 @@
+     * @throws TableNotFoundException
+     */
+    private List<Range> getIndexRanges(String indexTable,
+            Map<AccumuloColumnConstraint, Collection<Range>> constraintRangePairs,
+            Collection<Range> rowIDRanges)
+            throws TableNotFoundException
+    {
+        Set<Range> finalRanges = null;
+        // For each column/constraint pair
+        for (Entry<AccumuloColumnConstraint, Collection<Range>> e : constraintRangePairs
+                .entrySet()) {
+            // Create a batch scanner against the index table, setting the ranges
+            BatchScanner scan = conn.createBatchScanner(indexTable, auths, 10);
+            scan.setRanges(e.getValue());
+
+            // Fetch the column family for this specific column
+            Text cf = new Text(Indexer.getIndexColumnFamily(e.getKey().getFamily().getBytes(),
+                    e.getKey().getQualifier().getBytes()).array());
+            scan.fetchColumnFamily(cf);
+
+            // For each entry in the scanner
+            Set<Range> columnRanges = new HashSet<>();
+            for (Entry<Key, Value> entry : scan) {
+                entry.getKey().getColumnQualifier(tmpCQ);
+
+                // Add to our column ranges if it is in one of the row ID ranges
+                if (inRange(tmpCQ, rowIDRanges)) {
+                    columnRanges.add(new Range(tmpCQ));
+                }
+            }
+
+            LOG.info("Retrieved %d ranges for column %s", columnRanges.size(),
+                    e.getKey().getName());
+
+            // If finalRanges is null, we have not yet added any column ranges
+            if (finalRanges == null) {
+                finalRanges = new HashSet<>();
+                finalRanges.addAll(columnRanges);
+            }
+            else {
+                // Retain only the row IDs for this column that have already been added
+                // This is your set intersection operation!
+                finalRanges.retainAll(columnRanges);
+            }
+
+            // Close the scanner
+            scan.close();
+        }
+
+        // Return the final ranges for all constraint pairs
+        return new ArrayList<>(finalRanges);
+    }
+
+    /**
+     * Bins the given collection of ranges into metadata for the Presto splits
+     *
+     * @param numRangesPerBin Number of ranges to put in a single metadata object
+     * @param splitRanges The ranges to add
+     * @param prestoSplits The output collection for the tablet metadata
+     */
+    private void binRanges(int numRangesPerBin, List<Range> splitRanges,
+            List<TabletSplitMetadata> prestoSplits)
+    {
+        // Shuffle the handles to give an even distribution across all tablet splits
+        Collections.shuffle(splitRanges);
+
+        // Location here doesn't matter, I think this 9997 is left over from the connector example
+        // Let's leave it to pay homage to where this thing came from
+        String loc = "localhost:9997";
```

Comment:
Please remove it because it's misleading. If you want to pay homage to the code you took pattern from, put credits in README. :)

Is this particular case when you don't pay attention to `TableSplitMetadata.hostPort` or in general it's not necessary? If the latter, please remove it from `TabletSplitMetadata` AND here as well.